### PR TITLE
Increasing test coverage for toponetx

### PR DESCRIPTION
**Pull request by gurug-dev**
part of series of PRs for #246 

adding test cases and a minor fix for toponetx/utils/structure.py

### CODE REVIEW

**Review by mhajij — APPROVED** [review]

**Comment by ffl096** [comment]
@mhajij Did you mean to close this? Looks more like this was ready to be merged?

**Comment by gurug-dev** [comment]
@mhajij please reopen this PR. I had kept this as draft to add more commits related to TopoNetX test cases.

File: test/utils/test_structure.py [hunk 1]
```diff
@@ -141,3 +146,45 @@ def test_incidence_to_adjacency(self):
...
+        uidset = [10, 11, 12]
+        result = compute_set_incidence(children, uidset)
+        assert np.array_equal(
+            result.toarray(), np.array([[1, 1, 1], [1, 1, 1], [1, 1, 1]])
+        )
+
+        # Sparse matrix and index
+        children = [1, 2]
+        uidset = [10, 12]
+        result_ndict, result_edict, result_MP = compute_set_incidence(
```
Comment (ninamiolane) [inline test/utils/test_structure.py:163]: what does "MP" mean? 

Please avoid uppercase letters for variable names (python guidelines) --> result_mp instead.

**Review by ninamiolane — APPROVED** [review]
LGTM, just minor comment on variable name (no uppercase)

**Comment by gurug-dev** [comment]
marking it as draft for now, will be adding more commits.

File: test/utils/test_structure.py (hunk 1 above)
Comment (gurug-dev) [reply test/utils/test_structure.py:163]: will replace the variable name. MP is the name of the matrix given in the method, unsure on that, followed it as such assuming it to be notation.

File: test/classes/test_cell_complex.py [hunk 2]
```diff
@@ -956,6 +956,19 @@ def test_degree(self):
         assert CX.degree(7) == 2
         assert CX.degree(8) == 2
 
+        with pytest.raises(NotImplementedError):
+            CX.degree(node=1, rank=2)
+
+    def test_cell_neighbors(self):
+        """Test the cell neighbours of cell complex."""
+        CX = CellComplex()
```
Comment (mhajij) [inline test/classes/test_cell_complex.py:964]: change variable name to CC, everywhere

Comment (gurug-dev) [reply test/classes/test_cell_complex.py]: Done.

**Comment by gurug-dev** [comment]
Marking for review, will continue additional testing in a second PR. (trying to avoid conflicts with the Components refactor from Mustafa)

**Review by ninamiolane — APPROVED** [review]
Excellent, thanks!

**Pull request merged**